lib/resourcemerge/apps.go

```diff
@@ -3,14 +3,16 @@ package resourcemerge
 import (
 	appsv1 "k8s.io/api/apps/v1"
 	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/utils/pointer"
 )
 
 // EnsureDeployment ensures that the existing matches the required.
 // modified is set to true when existing had to be updated with required.
 func EnsureDeployment(modified *bool, existing *appsv1.Deployment, required appsv1.Deployment) {
 	EnsureObjectMeta(modified, &existing.ObjectMeta, required.ObjectMeta)
 
-	if required.Spec.Replicas != nil && *required.Spec.Replicas != *existing.Spec.Replicas {
+	ensureReplicasDefault(&required)
+	if existing.Spec.Replicas == nil || *required.Spec.Replicas != *existing.Spec.Replicas {
```

Comment:
You can have nil replicas in `required`, but I'm pretty sure API will always return at least `0` in `existing`. We probably don't need to update this line then

Reply:
Wait, nevermind, this would leave a risky `*existing.Spec.Replicas`, so it LGTM

Reply:
the `nil` guard for `existing` is mostly because we don't bother populating this in the unit-test mocks.  I would be very surprised if an API-server actually returned a resource with nil `replicas`.